cmake/gx-compile-shader.cmake

```diff
@@ -0,0 +1,22 @@
+FUNCTION(COMPILE_SHADER DIR SHADER ALL_INCLUDES SHADER_OUT)
+	IF (WIN32)
+		GET_FILENAME_COMPONENT(PARENT_DIR ${CMAKE_SOURCE_DIR} DIRECTORY)
+		GET_FILENAME_COMPONENT(FILE_NAME ${SHADER} NAME)
+		GET_FILENAME_COMPONENT(FILE_NAME_WO_EXT ${SHADER} NAME_WLE)
+
+		SET(SHADER_TYPE ${CMAKE_MATCH_1})
+		FIND_PROGRAM(DXC dxc PATHS ${GX_DXC_BINARY_DIR} REQUIRED NO_DEFAULT_PATH)
+
+		SET(OUTPUT_PATH ${PARENT_DIR}/assets/${FILE_NAME_WO_EXT}.cso)
```

Comment:
I think our decision, for now, was to place all of them beside the exe files.
Later we have to have a better way of handling our assets.